### ISSUE TITLE
The Deserters [Dustwallow Marsh quest]

### ISSUE THREAD

**Issue by Phatcat** [elysium-project/server#1941]
All bugged.

The deserters are friendly, they shouldn't be.

Balos Jecken is hostile, and never becomes friendly on low health, either he keeps resetting (evading, and then re-engaging in combat almost immediately after having regained a bit of health), or he is killed;

![billede](https://user-images.githubusercontent.com/1907820/28996784-c3f087aa-7a06-11e7-92c8-b7e8ad7e3625.png)

**Comment by kampeador** [comment]
Currently, this quest is implemented using DB only and is very buggy.
deserters were friendly pre 2.3.

**Comment by Phatcat** [comment]
> deserters were friendly pre 2.3.

Not when you challenge him on the quest, otherwise him saying 'we surrender' makes no sense at all. they may be friendly on approach, but they should turn hostile during the fight at least, which they don't.

**Comment by kampeador** [comment]
I haven't completed this quest pre 2.3, but there are some comments about this quest here:
https://web.archive.org/web/20060616173650/http://wow.allakhazam.com:80/db/quest.html?wquest=1286
There were some bugged quests in retail. I agree, it doesn't make sence at all.

You are proposing to change a quest logic. Keep in mind that classic wow has no phasing and there are a ton of cases to handle. Everything can happen with anything at any time.
Balos Jacken is initially hostile. On aggro, according to your proposition, all deserters must change their faction to hostile.
- case 1: Player is just an explorer and doesn't do any quests in this area? He rans into the tower, Balos Jacken aggros, all deserters also aggro, which will lead to "corpse running". It sounds like a trap.
- case 2: If we want to avoid case 1, then it is impossible to check, if an player has a quest, because other players, without it can help him.
- case 3: We need to safely track all deserters in the area, to ensure that their faction are correctly changed/restored. Some of them can spawn/respawn during the fight, map cell can be unloaded, etc.
- case 4: Players from the opposite faction(Horde) need to be handled separately.
There are alot of hidden complexity here, even if i implement all of these, i don't think it will be accepted.

Blizzard changed it in 2.3 along with many other quests. They made all deserters hostile.

**Comment by Phatcat** [comment]
> case 2: If we want to avoid case 1, then it is impossible to check, if an player has a quest, because other players, without it can help him.

Sounds like a restriction on our part, not on logic or possibilities. Are you telling me it's impossible to see who a player is in party with and if any of those players are on the quest? - Is there a core functionality for that? no. Could there be? Indeed.

**Comment by kampeador** [comment]
You need just to talk with Balos Jacken, you don't need to defeat him. You don't need to be in a group and it is not a group event at all.
On retail after 2.3, a random player, even from the opposite faction could defeat him. He says, that he is ready to cooperate(horde players couldn't understand him) and changes faction globally. Other random players, for a short period of time, can run in and just autocomplete this quest without fight. Deserters remain hostile all the time and were still attacking you even if you have defeated Balos Jacken. In the latest patches, this part was changed again: Once you defeat Balos Jacken, deserters that were attacking you, start fleeing and despawn.

**Comment by Phatcat** [comment]
> You just need to talk with Balos Jacken, you don't need to defeat him. You don't need to be in a group and it is not a group event at all.

No, yeah, I see the issue now.. tbh I don't know all the ins and outs of this quest, but it does sound tricky... no wonder blizzard changed it.. I just know that the current behaviour is bugged.

**Comment by Phatcat** [comment]
However, given this description;

> On retail after 2.3, a random player, even from the opposite faction could defeat him. He says, that he is ready to cooperate(horde players couldn't understand him) and changes faction globally. Other random players, for a short period of time, can run in and just autocomplete this quest without fight. Deserters remain hostile all the time and were still attacking you even if you have defeated Balos Jacken. In the latest patches, this part was changed again: Once you defeat Balos Jacken, deserters that were attacking you, start fleeing and despawn.

sounds very much like we don't wanna avoid case 1. it sounds like case 1 was indeed the correct implementation even in vanilla.

> case 3: We need to safely track all deserters in the area, to ensure that their faction are correctly changed/restored. Some of them can spawn/respawn during the fight, map cell can be unloaded, etc.

Simply watch them on their guid - which means this requires core and not db scripting, and on everyone who is alive removetempfaction/settempfaction ?

> case 4: Players from the opposite faction(Horde) need to be handled separately.

well, it's pretty easy, for horde players don't set/remove any temp faction since all deserters will already be hostile to them?

**Comment by Phatcat** [comment]
> Deserters remain hostile all the time and were still attacking you even if you have defeated Balos Jacken.

With that however, perhaps they only set temp faction but never removed it again. As you said, blizzard did have some bugged quests themselves, especially those involving scripts, and this would make perfect sense given the behaviour you describe as well as a the fact they had to change it to them fleeing and despawning. 

Either that or the fact that they changed the deserters to always be hostile was what broke them not being able to make them stop attacking you on having 'beat' Balos, Blizzard having to 'think out of the box' and introduce the fleeing and despawning thing as a fix.
